editor/src/clj/editor/sprite.clj

```diff
@@ -453,91 +496,111 @@
                                              :slice9 slice9
                                              :offset offset
                                              :playback-rate playback-rate
-                                             :attributes (graphics/attributes->build-target (pb-overridden-attributes material-attribute-infos vertex-attribute-overrides false))}
+                                             :attributes (build-target-attributes material-attribute-infos vertex-attribute-overrides vertex-attribute-bytes)}
                                             [:tile-set :material])]))
-(defn- fill-with-zeros [values expected-count]
-  (let [trailing-vec-with-zeros (repeat (- expected-count (count values)) 0)]
-    (vec (concat values trailing-vec-with-zeros))))
 
-(defn- get-attribute-property-type [attribute]
-  (let [attribute-element-count (:element-count attribute)
-        attribute-semantic-type (:semantic-type attribute)]
-    (cond (= attribute-semantic-type :semantic-type-color) types/Color
-          (= attribute-element-count 1) g/Num
-          (= attribute-element-count 2) types/Vec2
-          (= attribute-element-count 3) types/Vec3
-          (= attribute-element-count 4) types/Vec4)))
-
-(defn- get-attribute-expected-element-count [attribute]
-  (let [attribute-element-count (:element-count attribute)
-        attribute-semantic-type (:semantic-type attribute)]
-    (if (= attribute-semantic-type :semantic-type-color)
-      4
-      attribute-element-count)))
+(defn- attribute-property-type [attribute]
+  (case (:semantic-type attribute)
+    :semantic-type-color types/Color
+    (case (int (:element-count attribute))
+      1 g/Num
+      2 types/Vec2
+      3 types/Vec3
+      4 types/Vec4)))
+
+(defn- attribute-expected-element-count [attribute]
+  (case (:semantic-type attribute)
+    :semantic-type-color 4
+    (:element-count attribute)))
 
 (defn- attribute-update-property [current-property-value attribute new-value]
   (assoc current-property-value (:name-key attribute) new-value))
 
 (defn- attribute-clear-property [current-property-value attribute]
   (dissoc current-property-value (:name-key attribute)))
 
-(defn- get-attribute-edit-type [attribute prop-type]
-  (let [attribute-semantic-type (:semantic-type attribute)
+(defn- attribute-edit-type [attribute property-type]
+  (let [attribute-element-count (:element-count attribute)
+        attribute-semantic-type (:semantic-type attribute)
         attribute-update-fn (fn [_evaluation-context self _old-value new-value]
-                              (g/update-property self :vertex-attribute-overrides attribute-update-property attribute new-value))
+                              (let [values (if (= g/Num property-type)
+                                             (vector-of :double new-value)
+                                             new-value)]
```

Comment:
The `g/Num` property type expects a number, so we must unpack single-element attributes from their vector for editing using the property editor widget.